### PR TITLE
refactor(type): move sapphire-community/type

### PR DESCRIPTION
**Pull request by PyroTechniac**

### CODE REVIEW

File: packages/type/lib/addon.cc [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    }
+
+    auto isolate = args.GetIsolate();
+
+    v8::Local<v8::Promise> promise = args[0].As<v8::Promise>();
+
+    int state = promise->State();
+    v8::Local<v8::Value> values[2] = { v8::Integer::New(isolate, state) };
+    size_t number_of_values = 1;
+    if (state != v8::Promise::PromiseState::kPending)
```
Comment (Nytelife26) [inline packages/type/lib/addon.cc:15]: Can you use consistent bracing? Just a stylistic nitpick.

**Review by Nytelife26 — CHANGES_REQUESTED** [review]
Looks good on the C++ side of things, I'm not too familiar with the Node internals front so I'll let the core Sapphire team deal with that. Minor stylistic nitpick, though.

File: packages/type/package.json [hunk 2]
```diff
@@ -17,9 +15,14 @@
 	"scripts": {
 		"test": "jest",
 		"lint": "eslint src tests --ext ts --fix -c ../../.eslintrc",
-		"build": "rollup -c",
+		"build": "tsc -b src && yarn build-native",
+		"postbuild": "gen-esm-wrapper dist/index.js dist/index.mjs",
+		"prebuild": "yarn configure",
```
Comment (Favna) [inline packages/type/package.json:20]:
```suggestion
		"prebuild": "yarn configure",
		"build": "tsc -b src && yarn build-native",
		"postbuild": "gen-esm-wrapper dist/index.js dist/index.mjs",
```

Nitpick but pre->action->post reads better.

File: packages/type/src/index.ts [hunk 3]
```diff
@@ -1,11 +1,14 @@
...
+// const { getPromiseDetails, getProxyDetails } =
+// 	typeof process === 'undefined'
+// 		? {
+// 				getPromiseDetails: () => undefined,
+// 				getProxyDetails: () => undefined
+// 		  }
+// 		: // @ts-ignore process.binding exists
+// 		  process.binding('util');
+
+import bindings from 'bindings';
```
Comment (Favna) [inline packages/type/src/index.ts:10]:
```suggestion
import bindings from 'bindings';
```

Lets remove commented out code yeah ;)

File: packages/type/src/index.ts [hunk 4]
```diff
@@ -1,11 +1,14 @@
...
+const { getPromiseDetails, getProxyDetails }: typeof import('./addon') = bindings('addon');
```
Comment (Favna) [inline packages/type/src/index.ts:11]: Just wanna say the `typeof import('./addon')` type is kinda poggers

File: packages/type/tests/arrays.test.ts [hunk 5]
```diff
@@ -1,5 +1,7 @@
 import { Type } from '../src';
 
+console.log(process.cwd());
+
 describe('Arrays', () => {
```
Comment (Favna) [inline packages/type/tests/arrays.test.ts:5]:
```suggestion
import { Type } from '../src';

describe('Arrays', () => {
```

Woops

File: packages/type/tsconfig.json [hunk 6]
```diff
@@ -1,10 +1,9 @@
 {
 	"extends": "../ts-config/src/tsconfig.json",
 	"compilerOptions": {
-		"target": "ES2020",
-		"module": "ESNext",
 		"outDir": "build",
-		"rootDir": "src"
+		"rootDir": "src",
+		"importHelpers": true
```
Comment (Favna) [inline packages/type/tsconfig.json:6]:
```suggestion
		"rootDir": "src"
```

`importHelpers` is enabled by default

**Review by Favna — DISMISSED** [review]

File: packages/type/tsconfig.json (hunk 6 above)
Comment (PyroTechniac) [reply packages/type/tsconfig.json:6]: yeah, was fighting a bug at one point and tried it

**Review by Favna — APPROVED** [review]

**Review by Favna — APPROVED** [review]

File: package.json [hunk 7]
```diff
@@ -5,7 +5,7 @@
 		"packages/*"
 	],
 	"scripts": {
-		"clean": "rimraf packages/**/dist packages/**/node_modules/.cache",
+		"clean": "rimraf packages/**/dist packages/**/build packages/**/node_modules/.cache",
```
Comment (kyranet) [inline package.json:8]: Is this change intended?

**Review by kyranet — CHANGES_REQUESTED** [review]

File: package.json (hunk 7 above)
Comment (Favna) [reply package.json]: woopps, forgot to roll this back.

**Review by kyranet — APPROVED** [review]

**Review by Favna — APPROVED** [review]

**Review by vladfrangu — APPROVED** [review]
